xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/transactions/Wrappers.java

```diff
@@ -199,11 +228,44 @@ public static XrpCurrencyAmount ofDrops(UnsignedLong drops) {
      *
      * @return An {@link XrpCurrencyAmount} of the amount of drops in {@code amount}.
      */
-    public static XrpCurrencyAmount ofXrp(BigDecimal amount) {
-      if (FluentCompareTo.is(amount).notEqualTo(BigDecimal.ZERO)) {
-        Preconditions.checkArgument(FluentCompareTo.is(amount).greaterThanEqualTo(SMALLEST_XRP));
+    public static XrpCurrencyAmount ofXrp(final BigDecimal amount) {
+      Objects.requireNonNull(amount);
+
+      if (FluentCompareTo.is(amount).equalTo(BigDecimal.ZERO)) {
+        return ofDrops(UnsignedLong.ZERO);
+      }
+
+      // Whether positive or negative, clamp the amount to 0 if it's too small, or 100B if it's too big.
```

Comment:
@nkramer44 I need to fix this comment (the code isn't clamping right now, but is instead enforcing preconditions like the old code. I _think_ that's the right choice, in order to keep things functioning "as-is" but we've been bit so many times by invalid assumptions around numbers, I'll pose the question here.

Reply:
I think leaving preconditions here is ok for one main reason, which is that this method is never invoked during serialization, and will only be invoked by a developer. If, for some reason, they need to create an `XrpCurrencyAmount` that is smaller than the smallest XRP or larger than the largest XRP amount, they can just call `ofDrops` or `of`

Reply:
I'm going to capture this idea for future consideration in https://github.com/XRPLF/xrpl4j/issues/475 to unblock this PR (Leaving  unblock this PR (we can consider this change as part of that ticket's work).